### ISSUE TITLE
Touch Input Issues

### ISSUE THREAD

**Issue by dtynewydd** [microsoft/microsoft-ui-xaml#6291]
### Describe the bug

Since upgrading to Preview 3, WinUI 3 apps seem to be missing some touch functionality.

- Soft keyboard does not appear when tapping on text boxes. I've tried putting break points in the FocusEngaged and Tapped events but these are not triggered
- Long presses do not seem to register as right clicks. This means no context menus or even the visual feedback that a right click has occurred

### Steps to reproduce the bug

Soft keyboard:
1) Create a new application with the `[Experimental] Blank App, Packaged with WAP (WinUI 3 in Desktop)`
2) Remove existing Windows App SDK nuget packages
3) Install Microsoft.WindowsAppSDK  1.0.0-preview3
4) Add a textbox to the MainWindow
5) Deploy on a touch device
6) Try to enter text using soft keyboard

Long press:
7) Add a ContextFlyout to a control on the MainWindow
8) Deploy on a touch device
9) Long press on control with ContextFlyout attached

### Expected behavior

Soft keyboard should appear, allowing user to enter text

ContextFlyout should appear on long press

### IDE

Visual Studio 2019

### NuGet package version

Microsoft.WindowsAppSDK 1.0.0-preview3

### Project type

- Packaged

### Windows version

May 2021 Update (19043)

**Comment by scout208** [comment]
I'm having this same issue on the stable version of the Windows App SDK. Is there any workaround?

**Comment by jde-le** [comment]
Same here, critical issue, as we cannot use app in production

**Comment by cgladue** [comment]
Same here, this is causing real issues... Also i tried to launch TabTip.exe (The Touch Keyboard) When the textbox gets focus which works perfectly fine,  but when the control looses focus i cannot kill the keyboard it refuses to die without manually touching the X both elevated and non elevated cases.

**Comment by minze-it** [comment]
Same here! Extremely critical issue for our production app! Please fix this!

**Comment by byrialsen** [comment]
For me this is a major problem with a custom app that my company sold to several customers.

Any workarounds or at least any status on a fix for this. Please🙏

**Comment by byrialsen** [comment]
@gabbybilka: Is anyone looking into this.

This error makes it impossible to enter text into a Texbox on a Surface Tablet without hard-keyboard. 🙈🙈

**Comment by scout208** [comment]
@byrialsen During the community call last Wednesday I learned they have been investigating the issue and gotten closer to the source but don't have an ETA yet.

**Comment by byrialsen** [comment]
@scout208. I actually streamed the Community Call myself late in the evening. Maybe to late as I missed that this issues was mentioned 🙈. Thanks for the brush up 👍

**Comment by scout208** [comment]
@byrialsen It wasn't mentioned in the video so no surprise you missed it. I actually asked the question and got a response in the chat. Look in the chat around [36:02](https://youtu.be/fRpXbB8WUko?t=2164) in the video and you should see my question and Mike Hillberg's response.

**Comment by firth** [comment]
@gabbybilka Is there a public schedule? Do we know when 1.0.4 is possibly scheduled for release, and if this fix is still included? Will it be included in the 1.1 release?

I checked Microsoft.WindowsAppSDK version 1.1.0 Preview 3, which was released May 3, and this bug still wasn't fixed.

Thanks!

**Comment by An0d** [comment]
> @gabbybilka Is there a public schedule? Do we know when 1.0.4 is possibly scheduled for release, and if this fix is still included? Will it be included in the 1.1 release?
> 
> I checked Microsoft.WindowsAppSDK version 1.1.0 Preview 3, which was released May 3, and this bug still wasn't fixed.
> 
> Thanks!

Same here. Tried with WindowsAppSDK 1.1.0-preview3 on a Panasonic Toughbook CF-20 MK2
Still got the problem :(

**Comment by trojanmartin** [comment]
Any workaround for this issue?

**Comment by trojanmartin** [comment]
We also tried it, but winui does not support elevation and application is unable to start osk process.

**Comment by trojanmartin** [comment]
> > Any workaround for this issue?
> 
> We implemented a very dirty workaround by launching osk.exe when our TextBox base class gets focus but to be honest - that's so painful it's not usable in production...

We also tried it, but winui does not support elevation and application is unable to start osk process.

**Comment by firth** [comment]
I just tested the new stable/release version of Micrososft.WindowsAppSDK version 1.1.0, which was released on June 3, and this bug still wasn't fixed.

**Comment by firth** [comment]
I just tested the new stable/release version of Microsoft.WindowsAppSDK version 1.1.1, which was released yesterday on Tuesday, June 14, and this bug still wasn't fixed.

**Comment by minze-it** [comment]
The newly released version 1.0.4 contains the fix - but the version 1.1.1 doesn't. Why wasn't this fixed in both branches at the same time? And why are two versions parallel in active development when one is released for just 3 weeks?
Wouldn't it be better to swap from SDK 1.0 to 1.1 completely?

**Comment by firth** [comment]
> The newly released version 1.0.4 contains the fix - but the version 1.1.1 doesn't. Why wasn't this fixed in both branches at the same time? And why are two versions parallel in active development when one is released for just 3 weeks? Wouldn't it be better to swap from SDK 1.0 to 1.1 completely?

I just tested version 1.0.4 and it doesn't seem to fix this issue. The software keyboard does not open when a user taps on a TextBox control. We tested this with 1.0.4 on both Windows 11 and Windows 10 (including tablet mode).

**Comment by csforbes** [comment]
@minze-it , have you actually succeeded in using version 1.0.4 with "the fix?" I'm working on a touch-first application and am watching this issue intently.

**Comment by BenHarnett** [comment]
I created an SO question on this awhile ago. I would appreciate if it is fixed someone could update it there: https://stackoverflow.com/questions/71402782/touch-keyboard-not-appearing-in-winui-3-app

**Comment by chausner** [comment]
I am also affected by this bug. Would be great to have the fit in App SDK 1.1.x.

> This error makes it impossible to enter text into a Texbox on a Surface Tablet without hard-keyboard. 🙈🙈

It's still possible to enter text in this case by opening the onscreen keyboard manually but I agree it's an annoying workaround.

**Comment by devinduplessis** [comment]
My team is also waiting for this to be fixed. We have checked the latest version 1.1.2 and it still is not resolved

**Comment by krschau** [comment]
WinUI: https://task.ms/38130286
Input: https://task.ms/40885999

**Comment by firth** [comment]
> WinUI: https://task.ms/38130286 Input: https://task.ms/40885999

What does this mean? Are these links to Microsoft internal task tracking? Those of us outside Microsoft can't open those links.

It's still not fixed in Microsoft.WindowsAppSDK version 1.1.4, which was released on August 11th.

**Comment by krschau** [comment]
Yes, these are Microsoft internal tracking tasks. You'll see links like these sprinkled around, they're not the only way we link tasks but they can be pretty useful. Sorry if they confused anyone.

**Comment by penndai** [comment]
Is there any update for the touch screen issue? We also need this feature for the app running on surface.

**Comment by baaaaif** [comment]
> WinUI: https://task.ms/38130286 Input: https://task.ms/40885999

@krschau any updates on these tasks for people outside MS?

**Comment by karmelcorn** [comment]
This solution to manually get the input view when the textbox is focused  on [Stackoverflow](https://stackoverflow.com/questions/39618127/programmatically-open-on-screen-keyboard-in-uwp) seems to be a working workaround for now.

**Comment by minze-it** [comment]
When it'll not be in the 1.2 - which version will then contain the fix for this?
This is a pretty mandatory feature..

**Comment by brentafis** [comment]
Using `InputPanelInterop.GetForWindow` in conjunction with `InputPane.TryShow` seems to work.  First, you'll need to hook up the the `TexBox.GotFocus` event.

**Comment by AliGuemues** [comment]
@brentafis: can you provide a small example for your workaround?

**Comment by brentafis** [comment]
> @brentafis: can you provide a small example for your workaround?
> 
> Thank you

Using Visual Studio 2022 => New Project => Blank App, Packaged (WinUI3 in Desktop)

MainWindow.xaml
```
<Window
    x:Class="App4.MainWindow"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:App4"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d">

    <StackPanel Orientation="Horizontal" HorizontalAlignment="Center" VerticalAlignment="Center">
        <TextBox
            GotFocus="OnTextBoxGotFocus"
            Width="200"/>
    </StackPanel>
</Window>
```

MainWindow.xaml.cs
```
using Microsoft.UI.Xaml;
using WinRT.Interop;

namespace App4
{
    public sealed partial class MainWindow : Window
    {
        public MainWindow()
        {
            this.InitializeComponent();
        }

        private void OnTextBoxGotFocus(object sender, RoutedEventArgs e)
        {
            Windows.UI.ViewManagement.InputPane inputPane = Windows.UI.ViewManagement.InputPaneInterop.GetForWindow(WindowNative.GetWindowHandle(this));
            inputPane.TryShow();
        }
    }
}
```

**Comment by AliGuemues** [comment]
Thank you very much! But unfortanetly it is not working for me:

<img width="599" alt="image" src="https://user-images.githubusercontent.com/49402164/201343290-67cdd0f2-3668-43a9-9f1e-5117746064a2.png">

**Comment by brentafis** [comment]
@AliGuemues what type of project are you using/building?

**Comment by MohamedSherifm** [comment]
@AliGuemues I think you are trying to use this code in a page not a window, So you have to get a reference to your window so u can pass it to the GetWindowHandle().
U can get a reference to your window by making it internal in your App.xaml.cs.
![image](https://user-images.githubusercontent.com/53783285/201988910-3335d1fc-3398-41fa-85d1-7452c806700b.png)
then u can get the window from any part of your application like this.
![image](https://user-images.githubusercontent.com/53783285/201989172-5d4e38a7-0cc0-41fa-b2ff-d92b7a3acb42.png)

**Comment by AliGuemues** [comment]
@MohamedSherifm: I already tried  last week exactly what are you posted (after @baaaaif hint with the window) and still got exceptions. On App.xaml.cs I have a instance to the window, activated it on start and saved it as a property for the access from outside with App.Current.Window. I am not sure if the reason is the use of AppSDK Version 1.0.3 but unfortanetly I couldnt upgrade to a never version because we were ready to release with the project. I will try it later with the newest AppSDK version. I will update this post with the exact exception later.

At the end I sticked with the following solution:

https://stackoverflow.com/questions/38774139/show-touch-keyboard-tabtip-exe-in-windows-10-anniversary-edition

Long story short after this stackoverflow thread: I used the the nuget package Osklib (https://github.com/AlexeiScherbakov/osklib) and it now works really fine.

**Comment by nickidentivue** [comment]
For anyone arriving here from the MAUI issue that was closed, I found a workaround that is working for us on a Poindus POS system. It may well work with vanilla WinUI3 but haven't tested it.

For every Entry, Autocomplete, SearchBar control etc. where you require the touch screen keyboard to show (yes, I know, what a pain....), add a listener for the Focused and Unfocused events:

![image](https://user-images.githubusercontent.com/107918806/211799615-125469d4-2ac3-4bf4-9d3b-8fb92fa03bc3.png)

Then put this in the code behind:

![image](https://user-images.githubusercontent.com/107918806/211799760-bbb7ab9e-eb42-4dab-b2f8-601a419a49ac.png)

This assumes you have chosen the partial static class method for calling platform-specific code from cross-platform code as shown here: https://www.youtube.com/watch?v=fsO6pXz0umc

If you are using a different way, just add the appropriate version of this method to your Windows-specific code, remembering if you use the static partial method you'll need dummy functions in your Android/iOS/Tizen code that do nothing. 

![image](https://user-images.githubusercontent.com/107918806/211801408-87ec3654-af8b-443d-a96f-c57650ba8a17.png)

You'll also need using WinRT.Interop. This will also bring up the keyboard on non-touch devices so for your use case you may need some extra code to check for that.

**Comment by Wandtket** [comment]
For anyone still having issue with touch keyboard on tablets I've posted a universal fix for all textboxes, there's no need to modify each textbox it applies universally through App.xaml.

[Here's the link to the post](https://github.com/microsoft/WindowsAppSDK/discussions/3343)

**Comment by JanRajnoha** [comment]
Has this bug priority? Will it be mentioned and explained on the next WinUI Community call? This will be a problem for more and more people, just like, who can't release app to production.

@Wandtket Is normal to have pretty big delay between clicking (tapping) in TextBox and showing keyboard?

**Comment by JanRajnoha** [comment]
@brentafis I tried your solution, but it is not working. I have no error, but keyboard still not show.

@nickidentivue Does your solution support InputScope property? I tried it and if I want just numbers in TextBox limited by InputScope, where I could use Number value of enum, in your solution I only could use predefined, where for direct number input is Keyboard and Handwriting, which is not optimal for my customer. On the other hand, your code work instantly

**Comment by Wandtket** [comment]
> @Wandtket Is normal to have pretty big delay between clicking (tapping) in TextBox and showing keyboard?

@JanRajnoha There isn't a delay for me, are you using Single Project  MSIX for your Windows App SDK 1.2 project? If not that might do it in debug as that was causing massive slowdowns on our end. I think I might post a project with the example working next week.

**Comment by JanRajnoha** [comment]
Oh, okay, thank you, I'll try to play with it a little to find out more @Wandtket

**Comment by Wandtket** [comment]
> Oh, okay, thank you, I'll try to play with it a little to find out more @Wandtket 

I pasted it in my platform troubleshooting project.


https://github.com/Wandtket/Troubleshooting-Projects

**Comment by v4sh101** [comment]
If anyone is still struggling with this like myself. I have found that disabling new keyboard experience in the registry has allowed us to pull up the keyboard. The compromise with this is you must hit x on the keyboard to close it or a button.

The instructions:

    Select key [HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\TabletTip\1.7]
    Create value [New] ⇒ [DWORD (32 bit) value] 
    Change the value name to [DisableNewKeyboardExperience] 
    Change the name [DisableNewKeyboardExperience] Double-click to open the value data, change it to 1
     Restart the PC

**Comment by Wandtket** [comment]
@v4sh101 

This worked for me, however, it brings up the Windows 10 version of the TouchKeyboard and Microsoft may remove it / break it in some future update.

**Comment by v4sh101** [comment]
Agreed this is by no means a long term fix. Just an interim solution until we have a permanent solution in place.

**Comment by JanRajnoha** [comment]
Maybe long term, but extremely critical

**Comment by Scottj1s** [comment]
The internal issues noted above have been fixed and should be available in Windows App SDK 1.3

**Comment by RealGregor** [comment]
Are you sure that in the experimental version (1.3.230202101-experimental1) this issue is fixed? I've tried to package and publish an app and try it on a windows10 tablet in touch mode but the issue still persists?

**Comment by JanRajnoha** [comment]
I tried WASDK 1.3 Exp 1 and there it works okay, but it is okay, that for password box touch keyboard still not show?

**Comment by penndai** [comment]
Would like to share some experience below:
I also tried the Windows App SDK 1.3 experimental release.
It works sometimes but not stable, the first time when I click the textbox, the touch keyboard appears, but when I try another textbox, I am able to see the touch keyboard flashing, and no keyboard appear, and also I can see it tries to open Visual Studio (to debug something).
Also the password textbox not populate the touch keyboard.

**Comment by elliotdennis** [comment]
I tried the Windows App SDK 1.3 - Exp 1 and it fixed the touch keyboard, but doesn't work on password textboxes and the application breaks when using the DatePicker control.

**Comment by gabbybilka** [comment]
:tada:`Microsoft.WindowsAppSDK v1.3.0` has been released which fixes this issue.
* [Release Notes](https://aka.ms/windowsappsdk/1.3/1.3.0/release-notes)
* [Nuget](https://www.nuget.org/packages/Microsoft.WindowsAppSDK/1.3.230331000)

**Comment by JanRajnoha** [comment]
@gabbybilka Text input still not working for password box, so this issue should stay open

**Comment by gabbybilka** [comment]
@codendone @Scottj1s Are you able to repro the above?

**Comment by JanRajnoha** [comment]
In my case it is content dialog with password and text boxes in full screen or maximized app and there just text box is working (touch keyboard show), but password box not

**Comment by penndai** [comment]
Same for me, the password text box not working properly. 
The ticket shouldn't be closed !

**Comment by techjogger** [comment]
Updated to Windows App SDK 1.3 stable version, when XAML Number Boxes get Focus the Software keyboard is NOT launched.

**Comment by JanRajnoha** [comment]
Could we get some update on this? @gabbybilka

**Comment by WalkaZxamlem** [comment]
Hello, I also experience issue with PasswordBox, is there any update on that?

**Comment by JanRajnoha** [comment]
Could we get any update on this? @chigy

**Comment by JJBrychell** [comment]
This issue is being closed as the original behavior has been fixed.  This fix, however, did expose a different issue, with the same symptom, in Password box.  So that issue has been opened as #8946.
